sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/PubsubHelper.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.nexmark;
+
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubClient;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubClient.SubscriptionPath;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubClient.TopicPath;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubJsonClient;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubOptions;
+import org.apache.beam.sdk.util.BackOff;
+import org.apache.beam.sdk.util.BackOffUtils;
+import org.apache.beam.sdk.util.FluentBackoff;
+import org.apache.beam.sdk.util.Sleeper;
+import org.joda.time.Duration;
+
+/**
+ * Helper for working with pubsub and gcs.
+ */
+public class PubsubHelper {
+  /** Underlying pub/sub client. */
+  private final PubsubClient pubsubClient;
+
+  /** Project id. */
+  private final String project;
+
+  /** Topics we should delete on close. */
+  private final List<TopicPath> createdTopics;
+
+  /** Subscriptions we should delete on close. */
+  private final List<SubscriptionPath> createdSubscriptions;
+
+  /** How to sleep in retry loops. */
+  private final Sleeper sleeper;
+
+  /** How to backoff in retry loops. */
+  private final BackOff backOff;
+
+  private PubsubHelper(PubsubClient pubsubClient, String project) {
+    this.pubsubClient = pubsubClient;
+    this.project = project;
+    createdTopics = new ArrayList<>();
+    createdSubscriptions = new ArrayList<>();
+    sleeper = Sleeper.DEFAULT;
+    backOff = FluentBackoff.DEFAULT
+      .withInitialBackoff(Duration.standardSeconds(1))
+      .withMaxRetries(3)
+      .backoff();
+  }
+
+  /** Create a helper. */
+  public static PubsubHelper create(PubsubOptions options) throws IOException {
+    return new PubsubHelper(
+      PubsubJsonClient.FACTORY.newClient(null, null, options),
+      options.getProject());
+  }
+
+  /**
+   * Create a topic from short name. Delete it if it already exists. Ensure the topic will be
+   * deleted on cleanup. Return full topic name.
+   * @throws InterruptedException
```

Comment:
Do you mean throws `IOException` ?

Reply:
Good catch. Removed it from JavaDoc. It was copy error from existing Dataflow code.